### DISCUSSION TITLE
There is a way to keep track of the converted data?

### DISCUSSION THREAD

**Discussion by moacirrf** [vsch/flexmark-java#536] [General]
Hi,
First all, thank you for the awesome job that was made on Flexmark-java.

There is a way to keep track of the converted data?
For example
`# Title`
will generate
`<h1>Title</h1>`

I would like to know if have some way to store the both data, because i need to ensure that `<h1> Title </h1>` was generated from `# Title.`

I'm working on a [Netbeans Plugin](https://github.com/moacirrf/netbeans-markdown) and i would like to improve source and preview window.

Thanks

**Comment by moacirrf** [comment]
I found a way
```
        var options = new MutableDataSet();
        options.set(Parser.EXTENSIONS, asList(TablesExtension.create(),
                StrikethroughExtension.create(),
                TaskListExtension.create()))
                .set(HtmlRenderer.SOFT_BREAK, "<br />\n");

        var parser = Parser.builder(options).build();
        options.set(HtmlRenderer.SOURCE_POSITION_ATTRIBUTE, "md_source_position_attr");
        var renderer = HtmlRenderer.builder(options).build();
```
this will generate something like

`<h6 md_source_position_attr="7-11" >Text</h6>\n"`

7-11 are the offset on original md file.

thanks

**Comment by vsch** [comment]
@moacirrf, yes that is the way to do it. The library was originally created to have tracking of source position through the AST and to rendered HTML and re-formatted markdown. This is also the reason why it is slower than other implementations and may have a bit more overhead.

BTW, it is customary to set all options at the same time and just pass them where needed. Any options not used by a particular component will be ignored, in this case source position attribute of the HtmlReneder will be ignored by the parser.
